libs/providers/config-cat/src/lib/context-transformer.ts

```diff
@@ -0,0 +1,65 @@
+import { EvaluationContext, EvaluationContextValue } from '@openfeature/js-sdk';
+import { User as ConfigCatUser } from 'configcat-common/lib/RolloutEvaluator';
+
+function contextValueToString(contextValue: EvaluationContextValue): string | undefined {
+  if (typeof contextValue === 'string') {
+    return contextValue;
+  }
+
+  if (typeof contextValue === 'boolean' || typeof contextValue === 'number' || contextValue === null) {
+    return String(contextValue);
+  }
+
+  if (typeof contextValue === 'undefined') {
+    return contextValue;
+  }
+
+  if (contextValue instanceof Date) {
+    return contextValue.toISOString();
+  }
+
+  return JSON.stringify(contextValue);
+}
+
+function transformContextValues(contextValue: EvaluationContextValue): ConfigCatUser['custom'] | undefined {
+  if (contextValue === null) {
+    return undefined;
+  }
+
+  if (typeof contextValue !== 'object' || Array.isArray(contextValue)) {
+    const value = contextValueToString(contextValue);
+    return value ? { value } : undefined;
+  }
+
+  if (contextValue instanceof Date) {
+    return { value: contextValue.toISOString() };
+  }
+
+  return Object.entries(contextValue).reduce<ConfigCatUser['custom']>((context, [key, value]) => {
+    const transformedValue = contextValueToString(value);
+    return transformedValue ? { ...context, [key]: transformedValue } : context;
+  }, undefined);
+}
+
+function stringOrUndefined(param?: unknown): string | undefined {
+  if (typeof param === 'string') {
+    return param;
+  }
+
+  return undefined;
+}
+
+export function transformContext(context: EvaluationContext): ConfigCatUser | undefined {
+  const { targetingKey, email, country, ...attributes } = context;
+
+  if (!targetingKey) {
+    return undefined;
+  }
+
+  return {
+    identifier: targetingKey,
+    email: stringOrUndefined(email),
+    country: stringOrUndefined(country),
```

Comment:
I guess `email` and `country` are "first class" attributes in ConfigCat that get their own dedicated fields?

Reply:
Yes exactly. I also added that to the README now.